### DISCUSSION TITLE
What is the best practice for moving Location?

### DISCUSSION THREAD

**Discussion by ChadLee74** [openTCS/opentcs#38] [Q&A] [answered]
Hi, 
First and foremost, I'm really lucky that I meet openTCS. It seems I'm standing on shoulders of giants.

My business is container terminal where big Gantry Crane load or unload container from vessels and load or unload the containers on AGVs.
See below two images where the AGVs are and how Crane moves.
So far my understanding is the load or unload points should be locations which might map with peripherals devices.
Since crane is moving horizontally what is the best way to model the locations? (assuming system can get the coordination of crane during the operation)
Do I need to consider the Cranes are peripheral devices?

![AgvUnderCrane](https://user-images.githubusercontent.com/110446237/182300030-895d70a2-3fe8-47f8-bdb1-7c5a146894c8.png)
![CraneMove](https://user-images.githubusercontent.com/110446237/182300040-c4b595e0-dc06-4c38-8150-c64afd3aa8a5.png)
.

Look forward to hearing from you.

Best regards,
Chad

**Comment by swltr** [comment, marked as answer]
There are multiple solutions to this with openTCS:

1. Create a point and location in the plant model for every position that the crane can place a container on.
2. Create less points and locations in the plant model, and add a key-value property to the transport order that describes the offset of the container position relative to the destination point.

Solution 2 is less manual work when creating the plant model, but having less points also usually means less throughput, as openTCS will only let a single vehicle move to a destination point, even if the actual destination position is e.g. 20 metres away from the modeled point and the vehicles are shorter than that. Solution 1 is more work, but still often the better choice, IMO.
